### PR TITLE
Various imprvements for AC300 & AC500

### PR DESCRIPTION
**Pull request by AChep**
Recently got AC500 and has been experiencing [no battery packs issue](https://github.com/Patrick762/hassio-bluetti-bt/issues/109) and frequent Unavailable statuses. Spent a few hours debugging and improving stuff:
- added BMS / ARM / DSP versions;
- fixed an issue with the component failing to read >1 battery pack;
- fixed a crash if sensor is unavailable right from the start;
- wrote a simple script that creates polling ranges from fields -- yay, no need to do that manually! It also has some tolerance, because sending tons of commands takes time and the time is valuable;
- changed max packs number of AC500 to 6;
- increased polling timeout and interval to compensate for a possible 6 battery packs handling.

### CODE REVIEW

File: custom_components/bluetti_bt/bluetti_bt_lib/base_devices/ProtocolV1Device.py [hunk 1]
```diff
@@ -14,6 +14,8 @@ def __init__(self, address: str, type: str, sn: str):
         # Device info
         self.struct.add_string_field("device_type", 10, 6)
         self.struct.add_sn_field("serial_number", 17)
+        self.struct.add_version_field('arm_version', 23)
+        self.struct.add_version_field('dsp_version', 25)
```
Comment (Patrick762) [inline custom_components/bluetti_bt/bluetti_bt_lib/base_devices/ProtocolV1Device.py:18]: I'll try this with my EB3A, if it fails we just add it to the device itself

**Review by Patrick762 — COMMENTED** [review]
Looks good so far, I'll test it on my EB3A and AC300 before merging

**Comment by Patrick762** [comment]
Can you please rebase your branch?

**Comment by AChep** [comment]
@Patrick762 I merged the changes in the upstream to my branch

File: custom_components/bluetti_bt/bluetti_bt_lib/base_devices/ProtocolV1Device.py (hunk 1 above)
Comment (Patrick762) [reply custom_components/bluetti_bt/bluetti_bt_lib/base_devices/ProtocolV1Device.py:18]: Works on EB3A

File: custom_components/bluetti_bt/bluetti_bt_lib/field_attributes.py [hunk 2]
```diff
@@ -125,6 +125,14 @@ def __init__(
...
     "ac_output_power": PowerFieldAttributes("AC Output Power"),
     "dc_output_power": PowerFieldAttributes("DC Output Power"),
+    "dsp_version": FieldAttributes(
+        type=FieldType.NUMERIC,
+        name=f"DSP Version",
+    ),
+    "arm_version": FieldAttributes(
+        type=FieldType.NUMERIC,
+        name=f"ARM Version",
+    ),
```
Comment (Patrick762) [inline custom_components/bluetti_bt/bluetti_bt_lib/field_attributes.py:135]: Should be added to diagnostics, see const.py DIAGNOSTIC_FIELDS

**Review by Patrick762 — CHANGES_REQUESTED** [review]

**Comment by Patrick762** [comment]
just a small change, then we can merge

File: custom_components/bluetti_bt/bluetti_bt_lib/bluetooth/device_reader.py [hunk 3]
```diff
@@ -53,6 +53,8 @@ async def read_data(
         if filter_registers is not None:
             polling_commands = filter_registers
             pack_commands = []
+        _LOGGER.info("Pooling comands: " + ",".join([f"{c.starting_address}-{c.starting_address + c.quantity - 1}" for c in polling_commands]))
```
Comment (Patrick762) [inline custom_components/bluetti_bt/bluetti_bt_lib/bluetooth/device_reader.py:56]:
```suggestion
        _LOGGER.info("Polling commands: " + ",".join([f"{c.starting_address}-{c.starting_address + c.quantity - 1}" for c in polling_commands]))
```

File: custom_components/bluetti_bt/bluetti_bt_lib/devices/ac200m.py [hunk 4]
```diff
@@ -39,12 +40,9 @@ def pack_num_max(self):
 
     @property
     def polling_commands(self) -> List[ReadHoldingRegisters]:
-        return super().polling_commands + [
-            ReadHoldingRegisters(70, 5),
-            ReadHoldingRegisters(86, 3),
-            ReadHoldingRegisters(92, 1),
```
Comment (Patrick762) [inline custom_components/bluetti_bt/bluetti_bt_lib/devices/ac200m.py:45]: Sometimes I wonder why those registers were in the wrong place ...

**Comment by AChep** [comment]
Took me a while, but I resolved those :)

**Review by Patrick762 — APPROVED** [review]

**Pull request merged**